rust/src/owned/constraints.rs

```diff
@@ -34,3 +37,90 @@ impl<'a> From<ConstraintSystem<'a>> for ConstraintSystemOwned {
         owned
     }
 }
+
+impl From<Vec<((Vec<u64>, Vec<u8>), (Vec<u64>, Vec<u8>), (Vec<u64>, Vec<u8>))>> for ConstraintSystemOwned{
+    /// Creates a `ConstraintSystemOwned` from an R1CS vector
+    ///
+    /// # Examples
+    /// ```
+    ///  let vec: Vec<((Vec<u64>, Vec<u8>), (Vec<u64>, Vec<u8>), (Vec<u64>, Vec<u8>))> = vec![
+    ///         // (A ids values)  *  (B ids values)  =  (C ids values)
+    ///         ((vec![1], vec![1]), (vec![1], vec![1]), (vec![4], vec![1])),       // x * x = xx
+    ///         ((vec![2], vec![1]), (vec![2], vec![1]), (vec![5], vec![1])),       // y * y = yy
+    ///         ((vec![0], vec![1]), (vec![4, 5], vec![1, 1]), (vec![3], vec![1])), // 1 * (xx + yy) = z
+    ///     ];
+    ///
+    ///  let constraints = ConstraintSystemOwned::from(vec);
+    ///```
+
+    fn from(constraints: Vec<((Vec<u64>, Vec<u8>), (Vec<u64>, Vec<u8>), (Vec<u64>, Vec<u8>))>) -> ConstraintSystemOwned {
```

Comment:
This could take a reference `&[ … ]` instead of Vec.

Reply:
I updated the syntax. Please check that it is as intended.